public/hwcutils.h

```diff
@@ -87,6 +88,34 @@ inline OverlapType AnalyseOverlap(const hwcomposer::HwcRect<int>& rect,
   }
 }
 
+/**
+ * Pretty-print HwcRect for debugging.
+ */
+inline std::string StringifyRect(HwcRect<int> rect) {
+  std::stringstream ss;
+  ss << "{(" << rect.left << "," << rect.top << ") "
+     << "(" << rect.right << "," << rect.bottom << ")}";
+
+  return ss.str();
+}
+
+/**
+ * Pretty-print HwcRegion for debugging.
+ */
+inline std::string StringifyRegion(HwcRegion region) {
+  std::stringstream ss;
+  std::string separator = "";
+
+  ss << "[";
+  for (auto& rect : region) {
+    ss << separator << StringifyRect(rect);
+    separator = ", ";
+  }
+  ss << "]";
+
+  return ss.str();
+}
```

Comment:
Where do these get called from?

Reply:
These two are useful only while debugging, for dumping rects like display frame/surface damage... I was using them for convenience while adding logs.
Currently, it is not used anywhere, maybe we could use it when ENABLE_DISPLAY_DUMP is enabled, to print the source crop, surface damage and display frame?

Reply:
Seem reasonable. Just 1 additional nit, if these are just for debug printing I'd prefer to drop the 'inline' and let the compiler handle that for us.